static/src/javascripts/projects/common/modules/identity/consents.js

```diff
@@ -0,0 +1,281 @@
+// @flow
+
+import reqwest from 'reqwest';
+import fastdom from 'lib/fastdom-promise';
+import loadEnhancers from './modules/loadEnhancers';
+
+import { push as pushError } from './modules/show-errors';
+import {
+    addSpinner,
+    removeSpinner,
+    flip as flipCheckbox,
+    getInfo as getCheckboxInfo,
+} from './modules/switch';
+import { addUpdatingState, removeUpdatingState } from './modules/button';
+import {
+    getCsrfTokenFromElement,
+    getNewsletterHtmlPreferenceFromElement,
+    ERR_IDENTITY_HTML_PREF_NOT_FOUND,
+} from './modules/fetchFormFields';
+
+const submitPartialFormStatus = (
+    type: ?string = null,
+    formData: FormData
+): Promise<void> => {
+    const url = (() => {
+        switch (type) {
+            case 'consent':
+                return '/privacy/edit-ajax';
+            default:
+                throw new Error('Undefined form type');
+        }
+    })();
```

Comment:
Apologies, I haven't seen this in the first place: there is no reason why this has to be a function. Instead it can be a simple string.